internal/fields/validate.go

```diff
@@ -215,8 +215,38 @@ func initDependencyManagement(packageRoot string, specVersion semver.Version, im
 		return nil, nil, fmt.Errorf("can't create field dependency manager: %w", err)
 	}
 
+	//
+	// Check if the package embeds ECS mappings
+	//
+	packageEmbedsEcsMappings := buildManifest.ImportMappings() && !specVersion.LessThan(semver2_3_0)
+	if !packageEmbedsEcsMappings {
+		logger.Debugf("Package does not embed ECS mappings")
+	}
+
+	//
+	// Check if the stack version includes ECS mappings
+	//
+	stackIncludesEcsMapping := false
+
+	packageManifest, err := packages.ReadPackageManifestFromPackageRoot(packageRoot)
+	if err != nil {
+		return nil, nil, fmt.Errorf("can't read package manifest: %w", err)
+	}
+
+	if len(packageManifest.Conditions.Kibana.Version) > 0 {
+		kibanaConstraints, err := semver.NewConstraint(packageManifest.Conditions.Kibana.Version)
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid constraint for Kibana: %w", err)
+		}
+		stackIncludesEcsMapping = allVersionsIncludeECS(kibanaConstraints)
+	}
+
+	// If the package embeds ECS mappings, or the stack version includes ECS mappings, then
+	// we should import the ECS schema to validate the package fields against it.
 	var schema []FieldDefinition
-	if buildManifest.ImportMappings() && !specVersion.LessThan(semver2_3_0) && importECSSchema {
+	if (packageEmbedsEcsMappings || stackIncludesEcsMapping) && importECSSchema {
```

Comment:
Please, add or modify a test package so we have a test case with a package that uses ECS fields, but doesn't embeds the mappings.

Reply:
If we want to test the scenario where the package uses the ECS in Elasticsearch, we need a stack version 8.13.0. 

@mrodm, which version of the stack the tests run on? Can we have stack version 8.13.0 (for all tests or just one)?

Reply:
> In this implementation, the ECS fields are imported before validation by the test runners (e.g. kibana version condition is `^8.13.0`), so there should not be errors related to field definitions. Is it required to start an Elastic stack ^8.13.0 too? 

Correct! 

elastic-package validates the ECS fields using the JSON schema during the static checks, so it does not need to run on Elastic stack ^8.13.0.